File: AI/book/ch2/2-1/python/thing.py
# This is taken from https://github.com/aimacode/aima-python/blob/master/agents.py

class Thing:
    """This represents any physical object that can appear in an Environment.
    You subclass Thing to get the things you want. Each thing can have a
    .__name__  slot (used for output only)."""

    def __init__(self):
        self.state = 0

    def __repr__(self):
        return '<{}>'.format(getattr(self, '__name__', self.__class__.__name__))

    def is_alive(self):
        """Things that are 'alive' should return true."""
        return hasattr(self, 'alive') and self.alive

    def show_state(self):
        """Display the agent's internal state. Subclasses should override."""
        return self.state

    def display(self, canvas, x, y, width, height):
        """Display an image of this Thing on the canvas."""
        # Do we need this?
        pass

File: AI/book/ch2/2-1/javascript/thing.js
/*
	This represents any physical object that can appear in an Environment.
    You subclass Thing to get the things you want. Each thing can have a
    name variable (used for output only).
*/

class Thing {
	constructor () {
		this.state = 0;
	}

	isAlive(){
       return this.hasOwnProperty('alive') && this.alive == true;
	}

    showState(){
    	// Display the agent's internal state. Subclasses should override."""
        return this.state;
    }
        

    display(canvas, x, y, width, height){
    	// Display an image of this Thing on the canvas.
        // Do we need this?
        //pass;
    }
        
}

// Tests

var thing1 = new Thing();
console.log(thing1);
console.log(thing1.isAlive());

thing1.alive = true;
console.log('Made thing1 alive');
console.log(thing1.isAlive());
console.log(thing1);